auth/authenticators/basic-authenticator.ts

```diff
@@ -48,11 +48,11 @@ export class BasicAuthenticator extends Authenticator implements AuthenticatorIn
     this.password = options.password;
   }
 
-  public authenticate(options: AuthenticateOptions, callback: AuthenticateCallback): void {
+  public authenticate(options: AuthenticateOptions): Promise<void> {
     const authHeaderString = computeBasicAuthHeader(this.username, this.password);
```

Comment:
I don't know about TypeScript but with JS, you need to always wrap the code within a promise in order to catch errors.

```ts
public authenticate(options: AuthenticateOptions): Promise<void> { 
  return new Promise((resolve, reject) => {
     const authHeaderString = computeBasicAuthHeader(this.username, this.password); 
     const authHeader = { Authorization: authHeaderString } 
     options.headers = extend(true, {}, options.headers, authHeader);
     resolve();
 });
}
```

Reply:
Promises work the same in TS as they do JS. That is true, I just wasn't anticipating any errors here and wasn't handling them when I was using callbacks. But I am good with this approach here to be safe.

Reply:
I'm just trying to anticipate future changes to those methods that today don't return errors